contracts/voting-with-uuid/src/contract.rs

```diff
@@ -0,0 +1,516 @@
+}
+
+// finds the largest locked amount in participated polls.
+fn locked_amount(voter: &[u8], storage: &dyn Storage) -> Uint128 {
+    let token_manager = bank_read(storage).load(voter).unwrap();
+    token_manager
+        .locked_tokens
+        .iter()
+        .map(|(_, v)| *v)
+        .max()
+        .unwrap_or_default()
+}
+
+fn has_voted(voter: &Addr, a_poll: &Poll) -> bool {
+    a_poll.voters.iter().any(|i| i == voter)
+}
+
+pub fn cast_vote(
+    deps: DepsMut,
+    _env: Env,
+    info: MessageInfo,
+    poll_id: Uuid,
+    vote: String,
+    weight: Uint128,
+) -> Result<Response, ContractError> {
+    let poll_key = poll_id.as_bytes();
+    let state = config_read(deps.storage).load()?;
+
+
+    let mut a_poll = match poll(deps.storage).load(poll_key) {
+        Ok(poll) => poll,
+        Err(_) => return Err(ContractError::PollNotExist {}), 
+    };
+
+    if a_poll.status != PollStatus::InProgress {
+        return Err(ContractError::PollNotInProgress {});
+    }
+
+    if has_voted(&info.sender, &a_poll) {
+        return Err(ContractError::PollSenderVoted {});
+    }
+
+    let key = info.sender.as_str().as_bytes();
+    let mut token_manager = bank_read(deps.storage).may_load(key)?.unwrap_or_default();
+
+    if token_manager.token_balance < weight {
+        return Err(ContractError::PollInsufficientStake {});
+    }
+    token_manager.participated_polls.push(poll_id);
+    token_manager.locked_tokens.push((poll_id, weight));
+    bank(deps.storage).save(key, &token_manager)?;
+
+    a_poll.voters.push(info.sender.clone());
+
+    let voter_info = Voter { vote, weight };
+
+    a_poll.voter_info.push(voter_info);
+    poll(deps.storage).save(poll_key, &a_poll)?;
+
+    let attributes = vec![
+        attr("action", "vote_casted"),
+        attr("poll_id", poll_id.to_string()),
+        attr("weight", &weight),
+        attr("voter", &info.sender),
+    ];
+
+    let r = Response {
+        submessages: vec![],
+        messages: vec![],
+        attributes,
+        data: None,
+    };
+    Ok(r)
+}
+
+fn send_tokens(to_address: &Addr, amount: Vec<Coin>, action: &str) -> Response {
+    let attributes = vec![attr("action", action), attr("to", to_address.clone())];
+
+    Response {
+        submessages: vec![],
+        messages: vec![CosmosMsg::Bank(BankMsg::Send {
+            to_address: to_address.to_string(),
+            amount,
+        })],
+        attributes,
+        data: None,
+    }
+}
+
+
+pub fn make_uuid(
+    deps: DepsMut,
+    env: Env,
+    info: MessageInfo,
+) -> Result<Response, ContractError> {
+    let uuid = new_uuid(&env, deps.storage, deps.api)?;
+    let r = Response {
+        submessages: vec![],
+        messages: vec![],
+        attributes: vec![
+            attr("action", "make_uuid"),
+            attr("uuid", uuid.to_string()),
+        ],
+        data: None,
+    };
+    Ok(r)
+}
+
+pub fn make_seq_id(
+    deps: DepsMut,
+    env: Env,
+    info: MessageInfo,
```

Comment:
nit: unused value,deps -> _deps & env -> _ env & info -> _info ?

Reply:
changed it! thks